pkg/streamingpromql/types/data.go

```diff
@@ -47,9 +47,9 @@ func (i *InstantVectorSeriesDataIterator) Reset(data InstantVectorSeriesData) {
 // It returns the next point with the lowest timestamp.
 // If h is not nil, the value is a histogram, otherwise it is a float.
 // If no more values exist ok is false.
-func (i *InstantVectorSeriesDataIterator) Next() (t int64, f float64, h *histogram.FloatHistogram, ok bool) {
+func (i *InstantVectorSeriesDataIterator) Next() (t int64, f float64, h *histogram.FloatHistogram, hIndex int, ok bool) {
```

Comment:
What do you think about keeping this method as just returning the next values.

Then have a couple of extra methods to `getCurrentHIndex` / `getCurrentFIndex` ?

I feel like that would be neater and keep the methods clearer on their uses.

Reply:
The only places that use `Next` also need the histogram sample index, so adding an extra method for this seems unnecessary and possibly error prone (eg. imagine something called `Next` but used the previous value from `getCurrentHIndex`).

Reply:
Fair enough :+1: